haystack/modeling/model/optimization.py

```diff
@@ -119,20 +95,17 @@ def initialize_optimizer(
     :param distributed: Whether training on distributed machines
     :param grad_acc_steps: Number of steps to accumulate gradients for. Helpful to mimic large batch_sizes on small machines.
     :param local_rank: rank of the machine in a distributed setting
-    :param use_amp: Optimization level of nvidia's automatic mixed precision (AMP). The higher the level, the faster the model.
-                    Options:
-                    "O0" (Normal FP32 training)
-                    "O1" (Mixed Precision => Recommended)
-                    "O2" (Almost FP16)
-                    "O3" (Pure FP16).
-                    See details on: https://nvidia.github.io/apex/amp.html
+    :param use_amp: This option is deprecated. Only Pytorch automatic mixed precision (AMP) is supported. The Apex
+                    library is no longer supported. This means that `use_amp` is no longer used by this function
+                    since it is not needed to initialize native Pytorch AMP. If a value is provided a warning message
+                    will be raised.
     :return: model, optimizer, scheduler
     """
-    if use_amp and not AMP_AVAILABLE:
-        raise ImportError(
-            f"Got use_amp = {use_amp}, but cannot find apex. "
-            "Please install Apex if you want to make use of automatic mixed precision. "
-            "https://github.com/NVIDIA/apex"
+    if use_amp:
+        logger.warning(
+            "Only PyTorch automatic mixed precision is supported. The Apex library is no longer supported.\n"
+            "This means that use_amp is no longer used by modeling.model.initialize_optimizer since it is not needed\n"
+            "to initialize native PyTorch automatic mixed precision."
```

Comment:
A link to the documentation page about optimization should be added here, so that interested users can get more info. I'd say most important for them to know is that automatic mixed precision works without further ado if they pass `use_amp=True`.
I would suggest to only show the warning message if `use_amp` is a string. If it is a boolean, the user has changed the use of the parameter and already knows about the change.